### DISCUSSION TITLE
Fragility Function Database

### DISCUSSION THREAD

**Discussion by jnak2754** [NHERI-SimCenter/SimCenterCommon#223] [General Questions] [answered]
I found the fragility function databases for the HAZUS 5.1 Hurricane (from the [DamageAndLossModelLibrary](https://github.com/NHERI-SimCenter/DamageAndLossModelLibrary/tree/main)  and there are two sets of files, "fragility_fitted" and "fragility_unfitted." Can you provide more information of the difference in these two databases, if there is any?

**Comment by JustinBonus** [comment]
Thanks for your interest in using NHERI SimCenter tools! Apologies for the delayed response.

@zsarnoczay @jinyan1214 @fmckenna Any information that may be relevant to the question?

**Comment by zsarnoczay** [comment, marked as answer]
Thank you, @jnak2754 , for reaching out, and I apologize for the delayed response.

There are indeed two versions of the Hazus 5.1 hurricane fragility functions in the library. We are currently working on a dedicated documentation for the DLML that will provide detailed information about every dataset in the library. In the meantime, I can describe the main difference between the two datasets here:

The Hazus hurricane model uses piecewise linear functions to characterize the fragility functions of building archetypes. The probability of exceeding each damage state is stored for each 5 mph increment in the 50-250 mph wind speed domain. The raw files from Hazus are available [here](https://github.com/NHERI-SimCenter/DamageAndLossModelLibrary/tree/main/hurricane/building/portfolio/Hazus%20v5.1/data_sources/input_files). Storing fragility functions in this manner leads to less efficient calculations, but we wanted to preserve this approach for users who plan to do benchmark studies with the Hazus software, or just prefer to use these functions in their studies. The functions are available in the `fragility_unfitted` dataset.

The majority of piecewise linear functions in the Hazus library are very close to cumulative distribution functions (CDF) of Normal or Lognormal distributions. We reviewed all four damage states for all building archetypes and picked the best fitting Normal or Lognormal CDF for each. For the sake of transparency, we also provide the scripts used to perform this fitting [here](https://github.com/NHERI-SimCenter/DamageAndLossModelLibrary/blob/main/hurricane/building/portfolio/Hazus%20v5.1/data_sources/generate_library_files.py). The `fragility_fitted` dataset contains the same building archetypes, but now their fragility functions are Normal or Lognormal CDFs. These are easier and more efficient to use. It is also easier to explore similarities between various archetypes in this dataset because it is easier to compare the two parameters of a Normal/Lognormal CDF than the 40 parameters of the multilinear raw functions in Hazus.

An important caveat about the fitted functions: Although the majority of the raw data follows the Normal/Lognormal CDF, there are exceptions. In the online documentation for the DLML, we plan to publish a list of building archetypes where the fitted functions are poor approximations of the raw data. 
We have been collaborating with the Hazus team and their original contractor for the hurricane fragility development to better understand when the observed deviations are due to realistic behavior and when they are more likely due to some sort of artifacts in the original study.  As soon as we develop a solid understanding of this, we will share it with the community.

I hope this helps, please do not hesitate to ask if you have further questions.
